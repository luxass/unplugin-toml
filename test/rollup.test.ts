import { join } from "node:path";
import { rollup } from "rollup";
import { describe, expect, it } from "vitest";
import { testdir } from "vitest-testdirs";
import TOMLPlugin from "../src/rollup";
import { removeComments } from "./utils";

describe("rolldown", () => {
  it("expect toml import to be a json object", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic"));

    expect(testdirPath).toBeDefined();

    const bundle = await rollup({
      input: join(testdirPath, "basic.js"),
      plugins: [
        TOMLPlugin(),
      ],
    });
    const { output } = await bundle.generate({
      format: "esm",
      sourcemap: false,
    });

    expect(removeComments(output[0].code)).toMatchSnapshot();
  });

  it("expect toml import to be a string", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic"));

    expect(testdirPath).toBeDefined();

    const bundle = await rollup({
      input: join(testdirPath, "basic-raw.js"),
      plugins: [
        TOMLPlugin(),
      ],
    });
    const { output } = await bundle.generate({
      format: "esm",
      sourcemap: false,
    });

    expect(removeComments(output[0].code)).toMatchSnapshot();
  });

  it("handle transforms", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/transform"));

    expect(testdirPath).toBeDefined();

    const bundle = await rollup({
      input: join(testdirPath, "transform.js"),
      plugins: [
        TOMLPlugin({
          transform(data) {
            if (data != null && typeof data === "object" && "this" in data) {
              return {
                this: "transformed",
              };
            }
          },
        }),
      ],
    });
    const { output } = await bundle.generate({
      format: "esm",
      sourcemap: false,
    });

    expect(removeComments(output[0].code)).toMatchSnapshot();
  });
});
