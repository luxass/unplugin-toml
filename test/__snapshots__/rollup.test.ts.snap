// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rolldown > expect toml import to be a json object 1`] = `
"var data = {
  "pluginDir": "./plugins",
  "web": {
    "enabled": true
  },
  "logging": {
    "type": "stdout",
    "level": "info"
  }
};

export { data as config };
"
`;

exports[`rolldown > expect toml import to be a string 1`] = `
"var config = "pluginDir = \\"./plugins\\"\\n\\n[web]\\nenabled = true\\n\\n[logging]\\ntype = \\"stdout\\"\\nlevel = \\"info\\"\\n";

export { config };
"
`;

exports[`rolldown > handle transforms 1`] = `
"var data = {
  "this": "transformed"
};

export { data as config };
"
`;
